### DISCUSSION TITLE
ElectricSQL now compatible!

### DISCUSSION THREAD

**Discussion by gregzo** [capacitor-community/sqlite#476] [Show and tell]
Hello community,

Our team has high hopes for this local first, SQLite to Postgres sync engine:
https://github.com/electric-sql/electric

I built a database driver for compatibility with `capacitor-community/sqlite`:
https://github.com/electric-sql/electric/tree/main/clients/typescript/src/drivers/capacitor-sqlite

The ElectricSQL team is super responsive and doing great work. Do check them out!

**Comment by ygarg465** [comment]
I cannot use capacitor-sqlite in the electron platform even after installing 'better-sqlite3-multiple-ciphers' in electron package.json.
BTW I cannot use Vite's legacy plugin in vite.config.json as ElectricSQL is using BigInt internally which is not supported by legacy browsers.
This might be causing the problem.
Any insights on this?

**Reply by gregzo** [reply]
2 different issues there, RE Electron / Capacitor SQLite: haven't tried myself, but Capacitor SQLite advertises Electron support with some caveats - did you read the Electron Quirks blurb in the [README](https://github.com/capacitor-community/sqlite/blob/master/README.md)?

Also worth a read: https://github.com/capacitor-community/sqlite/blob/89ffbee450eea5152a30863c4dc8f3158446fab2/docs/ElectronBetterSQLite3.md

**Reply by ygarg465** [reply]
I temporarily resolved this problem by manually editing the rt/electron-plugins.js file which after `npx cap sync electron` adds CapacitorSQLitePlugin but it resolves the plugin from the project's root node_modules instead of electron's node_modules which was giving this error as `better-sqlite3-multiple-ciphers` is not present in parent node_modules but electron-plugins.js gets overridden when again running `npx cap sync electron`.

**Reply by ygarg465** [reply]
I managed to avoid this error by not converting electron-plugins.js file to ES6 module
